common/menu.c

```diff
@@ -27,24 +31,36 @@ static enum
     HBMENU_NETLOADER_ACTIVE,
     HBMENU_NETLOADER_ERROR,
     HBMENU_NETLOADER_SUCCESS,
+    HBMENU_THEME_MENU,
 } hbmenu_state = HBMENU_DEFAULT;
 
 void launchMenuNetloaderTask() {
     if(hbmenu_state == HBMENU_DEFAULT)
         if(netloader_activate() == 0) hbmenu_state = HBMENU_NETLOADER_ACTIVE;
 }
 
-void launchMenuBackTask()
-{
+void launchMenuBackTask() {
     if(hbmenu_state == HBMENU_NETLOADER_ACTIVE) {
         netloader_deactivate();
         hbmenu_state = HBMENU_DEFAULT;
-    } else {
+    }
+    else if(hbmenu_state == HBMENU_THEME_MENU) {
+        hbmenu_state = HBMENU_DEFAULT;
+        menuScan(rootPath);
+    }
+    else {
         menuScan("..");
     }
 
 }
 
+void launchApplyThemeTask(menuEntry_s* arg) {
+    const char* themePath = removeDriveFromPath(arg->path);
```

Comment:
If you use arg->path directly with current commit, the path written into the file doesn't contain "sdmc:" right?

Reply:
i am pretty sure it will.... the arg->path contains "sdmc:" and was caused my switch too crash a lot.... took me 12 restarts before i figured out what was going on...

Reply:
i mean... if you want.... you can change this on your switch.... and test it.... and see if it crashes it :3